### PR TITLE
Add synchronous replication docs

### PR DESCRIPTION
**Pull request by TomShawn**
### What is changed, added or deleted? (Required)

Add a document for the synchronous replication. Content migrated from https://github.com/tikv/tikv/pull/7740 and https://github.com/tikv/website/pull/211.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s): https://github.com/tikv/tikv/pull/7740 https://github.com/tikv/website/pull/211

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Have version specific changes 
- [ ] Might cause conflicts

### CODE REVIEW

File: pd-configuration-file.md [hunk 1]
```diff
@@ -375,3 +375,7 @@ Configuration items related to the [TiDB Dashboard](/dashboard/dashboard-intro.m
 + Determines whether to enable the telemetry collection feature in TiDB Dashboard.
 + Default value: `true`
 + See [Telemetry](/telemetry.md) for details.
+
+## `replication-mode`
+
+Configuration items related to the replication mode of a single Region. See [Enable synchronous replication in PD configuration file](/synchronous-replication.md#enable-synchronous-replication-in-pd-configuration-file) for details.
```
Comment (BusyJay) [inline pd-configuration-file.md:381]: It works for all regions instead of single one.

Comment (TomShawn) [inline pd-configuration-file.md:381]:
```suggestion
Configuration items related to the replication mode of all Regions. See [Enable synchronous replication in PD configuration file](/synchronous-replication.md#enable-synchronous-replication-in-pd-configuration-file) for details.
```

File: synchronous-replication.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+  "dr-auto-sync": {
+    "label-key": "zone",
+    "state": "sync"
+  }
+}
+```
+
+> **Note:**
+>
+> The replication mode indicates how a single Region is replicated, either `asynchronous` or `synchronous`. The replication state of the cluster indicates how all Regions are replicated, with the options of `async`, `sync-recover`, and `sync`.
```
Comment (TomShawn) [inline synchronous-replication.md:67]: @BusyJay I wrote it based on our discussion. PTAL whether I have any misunderstanding, thanks!

File: synchronous-replication.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+summary: Learn how to configure synchronous replication.
+---
+
+# Synchronous Replication for Dual Data Centers
+
+This document introduces how to configure synchronous replication for dual data centers.
+
+> **Warning:**
+>
+> Synchronous replication is still an experimental feature. Do not use it in a production environment.
```
Comment (lilin90) [inline synchronous-replication.md:12]:
```suggestion
> Synchronous replication is still an experimental feature. Do not use it in the production environment.
```

File: synchronous-replication.md [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+In the scenario of dual data centers, one is the primary center and the other is the DR (data recovery) center. When a Region has an odd number of replicas, more replicas are placed in the primary center. When the DR center is down for more than a specified period of time, the asynchronous mode is used by default for the replication between two centers.
+
+To use the synchronous mode, you can configure in the PD configuration file or change the replication mode manually using pd-ctl.
```
Comment (lilin90) [inline synchronous-replication.md:16]:
```suggestion
To use the synchronous mode, you can configure it in the PD configuration file or change the replication mode manually using pd-ctl.
```

File: synchronous-replication.md [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+
+## Enable synchronous replication in PD configuration file
+
+The replication mode is controlled by PD. You can configure in the PD configuration file when deploying a cluster. See the following example:
```
Comment (lilin90) [inline synchronous-replication.md:20]:
```suggestion
The replication mode is controlled by PD. You can configure it in the PD configuration file when deploying a cluster. See the following example:
```

File: synchronous-replication.md [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+dr-replicas = 1
+wait-store-timeout = "1m"
+wait-sync-timeout = "1m"
+```
+
+In the configuration above:
+
++ `dr-auto-sync` is the mode to enable synchronous replication.
++ The label key `zone` is used to distinguish different data centers.
++ TiKV instances with the `"z1"` value are considered the primary data center, and TiKV instances with `"z2"` are the DR data center.
```
Comment (lilin90) [inline synchronous-replication.md:41]:
```suggestion
+ TiKV instances with the `"z1"` value are considered in the primary data center, and TiKV instances with `"z2"` are in the DR data center.
```

File: synchronous-replication.md [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
++ `primary-replicas` is the number of replicas that should be placed in the primary data center.
++ `dr-replicas` is the number of replicas that should be placed in the DR data center.
++ `wait-store-timeout` is the time to wait before falling back to asynchronous replication.
++ `wait-sync-timeout` is the time to wait before forcing TiKV to change replication mode (currently not supported).
```
Comment (lilin90) [inline synchronous-replication.md:45]:
```suggestion
+ `wait-sync-timeout` is the time to wait before forcing TiKV to change the replication mode (currently not supported).
```

File: pd-configuration-file.md [hunk 8]
```diff
@@ -375,3 +375,7 @@ Configuration items related to the [TiDB Dashboard](/dashboard/dashboard-intro.m
 + Determines whether to enable the telemetry collection feature in TiDB Dashboard.
 + Default value: `true`
 + See [Telemetry](/telemetry.md) for details.
+
+## `replication-mode`
+
+Configuration items related to the replication mode of all Regions. See [Enable synchronous replication in PD configuration file](/synchronous-replication.md#enable-synchronous-replication-in-pd-configuration-file) for details.
```
Comment (TomShawn) [inline pd-configuration-file.md:381]:
```suggestion
Configuration items related to the replication mode of all Regions. See [Enable synchronous replication in PD configuration file](/synchronous-replication.md#enable-synchronous-replication-in-the-pd-configuration-file) for details.
```

**Comment by lilin90** [comment]
LGTM

File: synchronous-replication.md [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+
+In the configuration above:
+
++ `dr-auto-sync` is the mode to enable synchronous replication.
++ The label key `zone` is used to distinguish different data centers.
++ TiKV instances with the `"z1"` value are considered in the primary data center, and TiKV instances with `"z2"` are in the DR data center.
++ `primary-replicas` is the number of replicas that should be placed in the primary data center.
++ `dr-replicas` is the number of replicas that should be placed in the DR data center.
++ `wait-store-timeout` is the time to wait before falling back to asynchronous replication.
++ `wait-sync-timeout` is the time to wait before forcing TiKV to change the replication mode (currently not supported).
```
Comment (BusyJay) [inline synchronous-replication.md:45]: Perhaps we can just delete the description.

Comment (TomShawn) [reply synchronous-replication.md:45]: Do you mean `(currently not supported)`?

Comment (BusyJay) [reply synchronous-replication.md:45]: Entire description.

Comment (TomShawn) [reply synchronous-replication.md:45]:
```suggestion
```

File: synchronous-replication.md (hunk 2 above)
Comment (BusyJay) [reply synchronous-replication.md:66]: The first sentence is wrong. `mode` can only be `dr-auto-sync` or `majority`. I think you can just delete the sentence.

**Review by BusyJay — COMMENTED** [review]
rest lgtm

File: synchronous-replication.md (hunk 2 above)
Comment (TomShawn) [reply synchronous-replication.md:66]: Deleted.
```suggestion
> The replication state of the cluster indicates how all Regions are replicated, with the options of `async`, `sync-recover`, and `sync`.
```

**Review by lilin90 — APPROVED** [review]

**Pull request merged**